### ISSUE TITLE
Keep untracked files when merging

### ISSUE THREAD

**Issue by fniephaus** [hpi-swa/Squot#53]
Untrack files also need to be merged into a branch.
Currently, "untracked" changes on a feature branch get lost when the branch is merged.

**Comment by j4yk** [comment]
I would say this depends on #149 .

**Comment by j4yk** [comment]
# Workaround
For now you have to make sure that the current version in the Git browser already contains the non-Squeak files (such as a README) before you perform a merge (note: a merge also happens when you pull). To fix up a merge commit that deletes or reverts your non-Squeak files, do the following in the file system working copy with the command line Git client (before the push):

Let *commit-hash* be the hash of the latest commit that correctly contains your non-Squeak files.

    $ git checkout <commit-hash> -- <all top directories and file names that do not contain Squeak packages>
    $ git commit --amend

For example:
 
    $ git checkout abc1234 -- .travis.yml images/ README.md
    $ git commit --amend

If you have already created further commits after the merge (because you noticed the deleted/reverted files too late), you should leave out the --amend and, thus, create a new commit with a message like "Restore non-Squeak files inadvertently lost during the latest merge".

**Comment by LinqLover** [comment]
Another workaround if your local working copy is only one commit ahead the origin, which I personally prefer to CLI operations:

1. Copy the commit description of the newest local commit
1. Select the second latest commit and choose "Reset branch to this commit"
1. Pull
1. Commit your changes again, using the commit description from clipboard

**Comment by j4yk** [comment]
When you pull, the upstream changes to those external files are currently
lost if there is a real merge. So this only works because there is no real
merge, but a fast forward, and you actually do a manual rebase. Nice
observation.


Christoph Thiede schrieb am Mo., 4. Nov. 2019, 20:56:

> Another workaround if your local working copy is only one commit ahead the
> origin, which I personally prefer to CLI operations:
>
>    1. Copy the commit description of the newest local commit
>    2. Select the second latest commit and choose "Reset branch to this
>    commit"
>    3. Pull
>    4. Commit your changes again, using the commit description from
>    clipboard
>
>
